### PR TITLE
Upgrade Velero to v1.11.1

### LINKED ISSUE #1665: [2] Upgrade Velero to v1.10.3
**What Chart should be upgraded?**
upstream / velero: [vmware-tanzu/velero](https://github.com/vmware-tanzu/helm-charts/tree/main/charts/velero)
https://github.com/vmware-tanzu/velero/releases/tag/v1.10.3

https://github.com/vmware-tanzu/velero/compare/v1.10.2...v1.10.3

**Why should this Chart be upgraded?**

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**

- I checked the migration of the new Chart:
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by simonklb**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [x] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### Platform Administrator notice
Only perform the upgrade if no Velero backup or restore operation is active.
Upgrade the Velero client to v1.11.1 on your host machine.

### What does this PR do / why do we need this PR?

...

- Fixes #1665 

#### Additional information to reviewers

Tested on a local (kind) cluster.
* Deployed 1.10.2
* Created a backup from daily schedule
* Upgraded to 1.11.1
* Restored the backup
* Created a new backup from daily schedule
* Restored from the new backup
* Deployed Prometheus
* Checked target and a few metrics

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [x] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [x] The metrics are still exposed ~and present in Grafana after the change~
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Comment by simonklb** [comment]
I've now upgraded to v1.11.1 rather than v1.10.3 as v1.10.3 lacked an upstream Helm chart release.

Note that I also changed it from being a transparent change to a disruptive and added a platform admin notice as I can't guarantee that ongoing backups or restores won't fail if the upgrade is performed at the same time. I've not been able to reproduce it but I also don't have any proof that it's harmless. If anyone of you know otherwise please let me know.

**Review by viktor-f — COMMENTED** [review]
I'm also not sure if backupps and restores keep working during an upgrade. But it feels like an ok precaution to not backup or restore while upgrading.

I see the [latest version of the upstream helmchart](https://github.com/vmware-tanzu/helm-charts/blob/velero-5.2.0/charts/velero/Chart.yaml) has v1.12.2 of velero. Is is possible to upgrade directly to that one?

Also, I wonder if we should add a admin notice about downloading a newer version of the velero cli. What do you think about that?

**Comment by simonklb** [comment]
> I see the [latest version of the upstream helmchart](https://github.com/vmware-tanzu/helm-charts/blob/velero-5.2.0/charts/velero/Chart.yaml) has v1.12.2 of velero. Is is possible to upgrade directly to that one?

Repeating from offline meeting: I actively chose not to skip 1.11 as the [1.12 upgrade docs](https://velero.io/docs/v1.12/upgrade-to-1.12/) mentions that having 1.11 installed is a prerequisite. It's likely that upgrading directly to 1.12 would be fine but I think it's generally a good idea to not skip minor versions. Not only to avoid troubles during the upgrade process but also to make time to actually read through the changelog properly.

> Also, I wonder if we should add a admin notice about downloading a newer version of the velero cli. What do you think about that?

I think that is a very good idea. I did that when I tested things locally but for some reason it didn't strike me to also mention it here. Thanks for pointing it out, added!

**Review by viktor-f — APPROVED** [review]

**Comment by crssnd** [comment]
I installed the new version, but I have some ingress packets dropped for the velero pod.
Did you have that on your env, also?

**Review by crssnd — APPROVED** [review]
added a commit that should fix the netpol issues.
please check

**Comment by simonklb** [comment]
> I installed the new version, but I have some ingress packets dropped for the velero pod. Did you have that on your env, also?

Great catch! No I did not see this. I specifically tested to make sure that Prometheus was able to scrape it.

> added a commit that should fix the netpol issues. please check

I'm guessing this happened after rebasing on https://github.com/elastisys/compliantkubernetes-apps/commit/78d6723bd3399ef8dff1569d1a79eecab42a01eb

Please send the fix as a separate PR instead of including it here and include @aarnq as one of the reviewers.

**Review by crssnd — APPROVED** [review]

**Review by Pavan-Gunda — APPROVED** [review]

**Comment by aarnq** [comment]
> > I installed the new version, but I have some ingress packets dropped for the velero pod. Did you have that on your env, also?
> 
> Great catch! No I did not see this. I specifically tested to make sure that Prometheus was able to scrape it.
> 
> > added a commit that should fix the netpol issues. please check
> 
> I'm guessing this happened after rebasing on [78d6723](https://github.com/elastisys/compliantkubernetes-apps/commit/78d6723bd3399ef8dff1569d1a79eecab42a01eb)
> 
> Please send the fix as a separate PR instead of including it here and include @aarnq as one of the reviewers.

https://github.com/elastisys/compliantkubernetes-apps/pull/1960 added it here.

File: helmfile.d/values/velero-sc.yaml.gotmpl [hunk 1]
```diff
@@ -33,53 +33,34 @@ initContainers:
...
-  volumeSnapshotLocation:
-    config:
+    - bucket: {{ .Values.objectStorage.buckets.velero }}
+      prefix: service-cluster
       {{- if eq .Values.objectStorage.type "s3" }}
-      region: {{ .Values.objectStorage.s3.region }}
+      provider: aws
+      config:
+        region: {{ .Values.objectStorage.s3.region }}
+        s3ForcePathStyle: "true"
```
[the remark this answers is not in the document]
Comment (simonklb) [reply helmfile.d/values/velero-sc.yaml.gotmpl:46]: https://github.com/elastisys/compliantkubernetes-apps/pull/1961

**Pull request merged**